File: Maximum Subarray/maximum_subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if ( nums.length == 0 ){
            return 0;
        }
        else if ( nums.length == 1 ){
            return nums[0];
        }
        int max = nums[0];
        int localMax = nums[0];
        for ( int i = 1; i < nums.length; i++ ){
            localMax = Math.max(localMax + nums[i], nums[i]);
            max = Math.max(max, localMax);
        }
        return max;
    }
}

File: Maximum Subarray/Maximum_Subarray.kt
class Solution {
    fun maxSubArray(nums: IntArray): Int {
        var max = Int.MIN_VALUE
        if ( nums.size == 0 ){
            return max
        }
        else if ( nums.size == 1 ){
            return nums[0]
        }
        var i = 1
        var subMax = nums[0]
        max = Math.max(max, subMax)
        while ( i < nums.size ) {
            subMax = Math.max(nums[i], subMax + nums[i])      
            max = Math.max(max, subMax)
            i++
        }
        return max
    }
}